File: CSharoOOP/04-OOPPrinciplesPartOne/Animals/Animal.cs
using System;

namespace Animals
{
    public abstract class Animal : ISound
    {
        private string name;
        private string sex;
        private int age; 

        public void Sound()
        {
            Console.WriteLine("I dont know what I am...");
        }

        public int Age
        {
            get
            {
                return this.age;
            }
            private set
            {
                this.age = value;
            }
        }

        public string Sex
        {
            get
            {
                return this.sex;
            }
            private set
            {
                this.sex = value;
            }
        }

        public string Name
        {
            get
            {
                return this.name;
            }
            private set
            {
                this.name = value;
            }
        }


        public Animal(string name, int age, string sex)
        {
            this.Name = name;
            this.Age = age;
            this.Sex = sex;
        }
    }
}

File: JavaScript Applications/UnderscoreJS/scripts/classes/animal.js
(function () {
    define([], function () {

        var Animal = (function () {
            function Animal(name, species, legsCount) {
                this._name = name;
                this._species = species;
                this._legsCount = legsCount;
            }

            return Animal;
        }());

        return Animal;
    });
}());